tls/extensions/s2n_client_supported_groups.c

```diff
@@ -58,9 +58,38 @@ int s2n_recv_client_supported_groups(struct s2n_connection *conn, struct s2n_stu
     proposed_curves.data = s2n_stuffer_raw_read(extension, proposed_curves.size);
     notnull_check(proposed_curves.data);
 
-    if (s2n_ecc_evp_find_supported_curve(&proposed_curves, &conn->secure.server_ecc_evp_params.negotiated_curve) != 0) {
-        /* Can't agree on a curve, ECC is not allowed. Return success to proceed with the handshake. */
-        conn->secure.server_ecc_evp_params.negotiated_curve = NULL;
+    s2n_parse_client_supported_groups_list(&proposed_curves, conn->secure.mutually_supported_groups);
+    s2n_choose_supported_group(&conn->secure.server_ecc_evp_params, conn->secure.mutually_supported_groups);
+    return 0;
+}
+
+int s2n_parse_client_supported_groups_list(struct s2n_blob *iana_ids, struct s2n_ecc_evp_params *supported_groups) {
+    struct s2n_stuffer iana_ids_in = {0};
+
+    GUARD(s2n_stuffer_init(&iana_ids_in, iana_ids));
+    GUARD(s2n_stuffer_write(&iana_ids_in, iana_ids));
```

Comment:
This pattern exists in a lot of places in the code, but it ends up doing a memcpy on overlapping regions (blob.data to blob.data).

I would recommend using `s2n_stuffer_raw_write(&iana_ids_in, iana_ids.size)` because it avoids undefined behavior. Since the stuffer is local, you can see that it doesn't matter if that tainted flag is set.